### ISSUE TITLE
[IOS] Release build configuration causes arm64 architecture build error for GoogleMaps

### ISSUE THREAD

**Issue by FrancisLeigh** [react-native-maps/react-native-maps#3527]
### Is this a bug report?
Yes

### Have you read the [Installation Instructions](https://github.com/react-community/react-native-maps/blob/master/docs/installation.md)?
Yes

### Environment
```
    "react-native": "0.62.2",
    "react": "16.11.0",
    "react-native-maps": "0.27.1",
```

> I am using GoogleMaps for IOS
> Deployment Target IOS 11.0
> XCode Version 12.0 beta 4 (12A8179i)

### Process to recreate
1. Set your build configuration to *Release* on XCode 
2. Clean and build 
3. Witness Error

### Expected Behavior

Simulator to launch with app running in release mode.

### Actual Behavior

Error seen in logs:
```
ld: building for iOS Simulator, but linking in object file built for iOS, file '/Users/[username]/Documents/C/tonsr2/ios/Pods/GoogleMaps/Maps/Frameworks/GoogleMaps.framework/GoogleMaps' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### PODFILE 
```
platform :ios, '11.0'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

def add_flipper_pods!(versions = {})
  versions['Flipper'] ||= '~> 0.33.1'
  versions['DoubleConversion'] ||= '1.1.7'
  versions['Flipper-Folly'] ||= '~> 2.1'
  versions['Flipper-Glog'] ||= '0.3.6'
  versions['Flipper-PeerTalk'] ||= '~> 0.0.4'
  versions['Flipper-RSocket'] ||= '~> 1.0'

  pod 'FlipperKit', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitLayoutPlugin', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/SKIOSNetworkPlugin', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitUserDefaultsPlugin', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitReactPlugin', versions['Flipper'], :configuration => 'Debug'

  # List all transitive dependencies for FlipperKit pods
  # to avoid them being linked in Release builds
  pod 'Flipper', versions['Flipper'], :configuration => 'Debug'
  pod 'Flipper-DoubleConversion', versions['DoubleConversion'], :configuration => 'Debug'
  pod 'Flipper-Folly', versions['Flipper-Folly'], :configuration => 'Debug'
  pod 'Flipper-Glog', versions['Flipper-Glog'], :configuration => 'Debug'
  pod 'Flipper-PeerTalk', versions['Flipper-PeerTalk'], :configuration => 'Debug'
  pod 'Flipper-RSocket', versions['Flipper-RSocket'], :configuration => 'Debug'
  pod 'FlipperKit/Core', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/CppBridge', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FBCxxFollyDynamicConvert', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FBDefines', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FKPortForwarding', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitHighlightOverlay', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitLayoutTextSearchable', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitNetworkPlugin', versions['Flipper'], :configuration => 'Debug'
end

# Post Install processing for Flipper
def flipper_post_install(installer)
  installer.pods_project.targets.each do |target|
    if target.name == 'YogaKit'
      target.build_configurations.each do |config|
        config.build_settings['SWIFT_VERSION'] = '4.1'
      end
    end
  end
end

target 'tonsr' do
  # Pods for tonsr
  pod 'FBLazyVector', :path => "../node_modules/react-native/Libraries/FBLazyVector"
  pod 'FBReactNativeSpec', :path => "../node_modules/react-native/Libraries/FBReactNativeSpec"
  pod 'RCTRequired', :path => "../node_modules/react-native/Libraries/RCTRequired"
  pod 'RCTTypeSafety', :path => "../node_modules/react-native/Libraries/TypeSafety"
  pod 'React', :path => '../node_modules/react-native/'
  pod 'React-Core', :path => '../node_modules/react-native/'
  pod 'React-CoreModules', :path => '../node_modules/react-native/React/CoreModules'
  pod 'React-Core/DevSupport', :path => '../node_modules/react-native/'
  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  pod 'React-RCTAnimation', :path => '../node_modules/react-native/Libraries/NativeAnimation'
  pod 'React-RCTBlob', :path => '../node_modules/react-native/Libraries/Blob'
  pod 'React-RCTImage', :path => '../node_modules/react-native/Libraries/Image'
  pod 'React-RCTLinking', :path => '../node_modules/react-native/Libraries/LinkingIOS'
  pod 'React-RCTNetwork', :path => '../node_modules/react-native/Libraries/Network'
  pod 'React-RCTSettings', :path => '../node_modules/react-native/Libraries/Settings'
  pod 'React-RCTText', :path => '../node_modules/react-native/Libraries/Text'
  pod 'React-RCTVibration', :path => '../node_modules/react-native/Libraries/Vibration'
  pod 'React-Core/RCTWebSocket', :path => '../node_modules/react-native/'

  pod 'React-cxxreact', :path => '../node_modules/react-native/ReactCommon/cxxreact'
  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
  pod 'React-jsiexecutor', :path => '../node_modules/react-native/ReactCommon/jsiexecutor'
  pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
  pod 'ReactCommon/callinvoker', :path => "../node_modules/react-native/ReactCommon"
  pod 'ReactCommon/turbomodule/core', :path => "../node_modules/react-native/ReactCommon"
  pod 'Yoga', :path => '../node_modules/react-native/ReactCommon/yoga', :modular_headers => true

  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'

  # pod 'react-native-maps', path: '../node_modules/react-native-maps'
  # pod 'react-native-google-maps', path: '../node_modules/react-native-maps'  # Remove this line if you don't want to support GoogleMaps on iOS
  # pod 'GoogleMaps'  # Remove this line if you don't want to support GoogleMaps on iOS
  # pod 'Google-Maps-iOS-Utils' # Remove this line if you don't want to support GoogleMaps on iOS

  # React Native Maps dependencies
  # https://github.com/react-native-community/react-native-maps/blob/master/docs/installation.md#enabling-google-maps-on-ios-react-native-all-versions
  rn_maps_path = '../node_modules/react-native-maps'
  pod 'react-native-google-maps', :path => rn_maps_path
  pod 'GoogleMaps'
  pod 'Google-Maps-iOS-Utils'

  pod 'react-native-netinfo', :path => '../node_modules/@react-native-community/netinfo'

  pod 'appcenter-core', :path => '../node_modules/appcenter/ios/appcenter-core.podspec'

  target 'tonsrTests' do
    inherit! :complete
    # Pods for testing
  end

  use_native_modules!

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable these next few lines.
  add_flipper_pods!
  post_install do |installer|
    flipper_post_install(installer)

    # https://developer.apple.com/forums/thread/656509
    # installer.pods_project.targets.each do |target|
    #   target.build_configurations.each do |config|
    #     config.build_settings['ONLY_ACTIVE_ARCH'] = 'YES'
    #   end
    # end
  end
end

target 'tonsr-tvOS' do
  # Pods for tonsr-tvOS

  target 'tonsr-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end
end


```

**Comment by FrancisLeigh** [comment]
@christopher-18 nothing resolved yet. have you had any luck?

**Comment by awinograd** [comment]
This started for me after upgrading to xcode 12

**Comment by frodoe7** [comment]
> This started for me after upgrading to Xcode 12

Me too , anyone reached to any solution ?

**Comment by chunterb** [comment]
A temporary workaround is to add arm64 to the excluded architectures within Xcode.

Xcode > Build Settings > Architectures > Excluded Architectures => Add "arm64"

<img width="706" alt="Screen Shot 2020-09-18 at 3 37 26 PM" src="https://user-images.githubusercontent.com/31895576/93643138-fce51800-f9c4-11ea-88ce-7b13ad59767c.png">

**Comment by HugoGresse** [comment]
Previous solution is not compatible for both simulator and distribution. 

After reading [this thread](https://developer.apple.com/forums/thread/656509), here is what you need to do: 

>  the issue appears to be Xcode 12's inclusion of arm64 arch in the iPhone simulator, thus causing a clash between iPhone and Simulator when they are being combined to create the fat library.

*Solution* (at least until simulator run on arm64 macs, which is not very soon)
Select the project or target -> Build Settings -> Excluded Architecture (if you can't see it make sure [All] properties are selected) and set the Excluded Architecture for 'Any iOS simulator SDK' -> 'arm64'.

<img width="654" alt="Capture d’écran 2020-09-21 à 16 59 51" src="https://user-images.githubusercontent.com/662377/93783571-218ef900-fc2c-11ea-85da-14646ea4f984.png">

**Comment by awinograd** [comment]
Thanks @HugoGresse ! Your solution worked perfectly for my app.

**Comment by kidmysoul** [comment]
@HugoGresse 
I followed your steps but these errors happened.
![スクリーンショット 2020-09-23 19 35 32](https://user-images.githubusercontent.com/15664045/94001822-08c14780-fdd4-11ea-9652-5824a2f95c13.png)

**Comment by christopher-18** [comment]
@kidmysoul  You can look into this, it may help you.
https://github.com/react-native-community/react-native-maps/issues/3536

**Comment by yarapolana** [comment]
In my work computer xcode updates automatically, so all release builds stopped working with xcode 12 so did freshly created rn apps, the release builds fail as well. thanks @HugoGresse for the solution.

**Comment by diegoflorez** [comment]
@HugoGresse solution works for me! Thanks

**Comment by Florian-crg** [comment]
facing the same problem. 
@HugoGresse's solution dosen't work for me on Apple M1. 
I got new errors about 
```
Showing All Issues
Undefined symbol: _CDV...
Undefined symbol: _OBJC...
```

**Comment by HugoGresse** [comment]
well, that's seems logical... Good luck!

**Comment by MujtabaFR** [comment]
Excluding architecture arm64 doesn't work for **M1 macs**
Any other solution ?

**Comment by MujtabaFR** [comment]
The problem is still active in [google-maps-ios-utils](https://github.com/googlemaps/google-maps-ios-utils/issues/355)

For **M1 Macs**: running Xcode under **Rossetta** will fix the issue as a temporary workaround 
```
1 - With Xcode closed (Important) Go to finder -> Applications
2 - Right Click on Xcode and select "Get Info"
3 - On the info panel check "Open using Rosetta"
4 - Double Click in the bottom large preview of the info panel.
```
** This will cause major decrease in speed and performance in Xcode and simulator

**Comment by Snowsoul** [comment]
I can confirm @MujtabaFR. 

This seems to be a bigger issue on m1 where excluding arm64 doesn't fix the problem. 
The only workaround for this is to run Xcode under Rossetta which is a shame.

Hope this will get fixed soon so we don't run Xcode in limp mode.

**Comment by salah-ghanim** [comment]
from my point of view this is off topic mainly because the bug is in a different repo (GoogleMaps) and we need to be patient until a new release is made then we can update our dependencies.

[https://github.com/googlemaps/google-maps-ios-utils/issues/355](GoogleMapsIOS issue)

**Comment by chiragbaraliya** [comment]
> ### Temporary Solution 1
> Xcode > Build Settings > Architectures > Excluded Architectures > Add "arm64"
> 
> <img alt="Screen Shot 2020-09-18 at 3 37 26 PM" width="706" src="https://user-images.githubusercontent.com/31895576/93643138-fce51800-f9c4-11ea-88ce-7b13ad59767c.png">
> 
> ### Temporary Solution 2
> Add the following to your iOS Podfile.
> 
> ```
> post_install do |installer_representation|
>     installer_representation.pods_project.targets.each do |target|
>         target.build_configurations.each do |config|
>             config.build_settings['ONLY_ACTIVE_ARCH'] = 'NO'
>         end
>     end
> end
> ```
> 
> **Note: This should NOT be a permanent solution.**

Awesome!! It works!!

**Comment by inoffice** [comment]
Use  in PODfile

platform :ios, '11.0'
use_frameworks!
inhibit_all_warnings!

target 'test' do
  pod 'GoogleMaps'
end

see in https://github.com/googlesamples/google-services/issues/496

BEST REGARDS

**Comment by jwaldrip** [comment]
@salah-ghanim I did not mean to be insulting. I have updating my language to help lighten the comment.

**Comment by n76i** [comment]
still not working in M1 machines. :(

**Comment by Shaninnik** [comment]
@salah-ghanim sorry to comment on the closed issue, but google maps just released v6.0 that they claim should work with M1 Macs:

https://cloud.google.com/blog/products/maps-platform/announcing-version-60-maps-and-places-sdks-ios

Is it realistic to expect an update soon?

**Comment by yonitou** [comment]
Hi there, 

Completely agreed with @Shaninnik. We've been patient for GoogleMaps to update their SDK for iOS. 

Can we reopen this issue and update this dependency to make it accessible to all react-native users ? 

Thanks a lot !

**Comment by hisamafahri** [comment]
I am experiencing this issue:

```
"react": "17.0.2",
"react-native": "0.66.3",
"react-native-maps": "^0.29.4"
```

**Comment by tolik85** [comment]
How come Google is not fixing such a bug? 
There any many M1 macs out there now...

**Comment by rushelex** [comment]
> How come Google is not fixing such a bug? There any many M1 macs out there now...

This is a react-native-maps problem, not Google's

**Comment by hisamafahri** [comment]
> This is a react-native-maps problem, not Google's

@rushelex It is Google's issue, but as @Shaninnik reply, that [Google released v6.0](https://cloud.google.com/blog/products/maps-platform/announcing-version-60-maps-and-places-sdks-ios), it might shine a new hope to solve this issue.

But as mentioned in the v6.0 announcement:

```
... XCFramework versions of the Maps and Places SDKs are still in beta, ...
```

this issue is involving both Google and react-native-maps.

**Comment by tolik85** [comment]
Does this mean that any project that uses Google Maps is unable to run on a simulator?

**Comment by hisamafahri** [comment]
> Does this mean that any project that uses Google Maps is unable to run on a simulator?

I do believe not all of them. As far as I know, this issue is mostly occur if you are using iOS simulator in the Apple M1 machines.

**Comment by tolik85** [comment]
Which is me, and I believe a lot of people since MacBook Pro 16" with M1 Max is available worldwide :)

**Comment by salah-ghanim** [comment]
@tolik85 can you please check against master now ? the latest Google Maps SDK for iOS (6.0.1) I can't find a release note for that version but I imagine the issue is fixed

**Comment by SakeFabi** [comment]
Does anyone have some good news? If google fixed this, will it immediately translate into being fixed in RN?

**Comment by jwaldrip** [comment]
Any news on this?

**Comment by Simon-TechForm** [comment]
@jwaldrip could you try installing from master and report back? `react-native-maps` now depends on the newest `GoogleMaps` podspec as of #4061.

**Comment by saeedtkh** [comment]
any news? I have the same issue!

**Comment by Shaninnik** [comment]
To anyone having problems with running on simulator on M1 macs and not wanting to wait for new release, just install this lib from git master branch. I can 100% confirm that it works, you might or might not need to tweak `EXCLUDED_ARCHS` in XCode, for me setting `"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "i386 arm64"` and 
```
installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
        config.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = 'i386 arm64'
      end
    end
``` 
in pod file worked. However there is one gotcha (at least for me) with `onMapReady` described here https://github.com/react-native-maps/react-native-maps/issues/4066 Until it is fixed I was able to work around it by using `onMapLoaded` instead

**Comment by Simon-TechForm** [comment]
Here's an update on the current situation of this issue:

Installing from `master` isn't necessary as I've released a beta version of `react-native-maps` on the `@beta` channel of npm that includes the updated version of the Google Maps SDK, that everyone expected to include a fix for M1 macs.

But for whatever reason, Google has decided not to include the fix in the CocoaPods release of the library (see [this](https://github.com/googlemaps/google-maps-ios-utils/issues/355#issuecomment-1060959728)). I have no insights as to why this is, but until the M1 fix is released in the CocoaPods version, there's not much we can do about it.

I suggest that someone who is affected by this opens an issue on [Google's public issue tracker](https://issuetracker.google.com/issues?q=status:open%20componentid:188833&s=created_time:desc) and everyone else affected supports this.

**Comment by russmenum** [comment]
Is this the same [issue](https://github.com/react-native-maps/react-native-maps/issues/4183)?

```
Installing Google-Maps-iOS-Utils (3.10.3)
Installing GoogleMaps (5.1.0)
Installing react-native-google-maps (0.30.1)
```

```
  "Swift.Array.init() -> [A]", referenced from:
      variable initialization expression of GoogleMapsUtils.HeatMapInterpolationPoints.(data in _085340145A7263942932318178DB6148) : [__C.GMUWeightedLatLng] in libGoogle-Maps-iOS-Utils.a(HeatMapInterpolationPoints.o)
      variable initialization expression of GoogleMapsUtils.HeatMapInterpolationPoints.(heatMapPoints in _085340145A7263942932318178DB6148) : [__C.GMUWeightedLatLng] in libGoogle-Maps-iOS-Utils.a(HeatMapInterpolationPoints.o)
      GoogleMapsUtils.HeatMapInterpolationPoints.init(givenClusterIterations: Swift.Int) -> GoogleMapsUtils.HeatMapInterpolationPoints in libGoogle-Maps-iOS-Utils.a(HeatMapInterpolationPoints.o)
      GoogleMapsUtils.HeatMapInterpolationPoints.(kcluster in _085340145A7263942932318178DB6148)() -> [[__C.CLLocationCoordinate2D]] in libGoogle-Maps-iOS-Utils.a(HeatMapInterpolationPoints.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

```
I tried this exclude arm64, and this is an Intel not M1 Mac

<img width="679" alt="Screen Shot 2022-04-12 at 10 05 53 AM" src="https://user-images.githubusercontent.com/77687831/162980962-f66b71f9-457a-4799-9cf3-340eadc94aa4.png">

**Comment by russmenum** [comment]
@Simon-TechForm since you forced closed my issue, will. you at least confirm if this is the same issue when it won't build on an Intel Mac?

**Comment by whysetiawan** [comment]
My issue was solved by doing these step :
1. install Rosseta using the terminal
`softwareupdate --install-rosetta`
2. Go to Applications -> Right Click Xcode -> Select "Get Info" -> check the "Open using rosseta"
3. Open Xcode and `clean` and `build` the app via Xcode

**Note** : i was using `"react-native-maps": "^0.31.0-beta.5"`

**Comment by jwaldrip** [comment]
@whysetiawan building under rosetta is not solving for ARM builds. It is a workaround, but this has other implications for many other users on m1 macs.

**Comment by jwaldrip** [comment]
@russmenum using the latest beta build will fix this but as @Shaninnik mentioned, there is an issue with the onMapReady function. If you are not using that, there should not be an issue.

**Comment by russmenum** [comment]
> @russmenum using the latest beta build will fix this but as @Shaninnik mentioned, there is an issue with the onMapReady function. If you are not using that, there should not be an issue.

Are you referring to my not force closed ticket? If so how do you install the beta instead of the main?

**Comment by whysetiawan** [comment]
> > @russmenum using the latest beta build will fix this but as @Shaninnik mentioned, there is an issue with the onMapReady function. If you are not using that, there should not be an issue.
> 
> @jwaldrip Are you referring to my not force closed ticket? If so how do you install the beta instead of the main? or just manually change the JSON to be `"react-native-maps": "^0.31.0-beta.5",`?

`npm install react-native-maps@beta`

**Comment by russmenum** [comment]
Can I ask, what did everyone do to get ```pod install``` using google maps to even run right on an M1 Mac?

If I ```cd ios``` and ```pod install``` with 

```
# React Native Maps dependencies
  rn_maps_path = '../node_modules/react-native-maps'
  pod 'react-native-google-maps', :path => rn_maps_path
```

it will not even finish the pod install, instead of a crapload of errors. Willing to try the run in rosetta, as have everything working on the Intell Mac, but there is no point to try and build before ```pod install``` successfuly completes

**Comment by salah-ghanim** [comment]
**working solution**
regarding M1 , iOS simulator and latest version of google maps this is a clear description of the current status and a working solution:
https://github.com/googlemaps/google-maps-ios-utils/issues/355#issuecomment-1060959728


TLDR use the 6.0.1-beta as an [XCFramework](https://developers.google.com/maps/documentation/ios-sdk/config#install-the-xcframework).
  * ether use carthage instead of cocoapods to import google maps
  * manually import the google maps ios utils

**Comment by jpamarohorta** [comment]
Hello @salah-ghanim, I'm trying to manually install and use the Google Maps SDK XCFramework but I'm running into multiple issues. Can you provide some guidance on how to make react-native-maps use the XCFramework instead of the dependency installed via cocoapods? Thank you!

**Comment by Simon-TechForm** [comment]
Hello everyone!

I've just released a `1.4.0` that should fix the m1/arm build issues reported here. Please check it out :)

**Comment by mikezamayias** [comment]
This fixed issue on my end: https://github.com/googlemaps/google-maps-ios-utils#cocoapods

**Comment by nitinparwani** [comment]
My issue was fixed by following steps :
Machine : M2 Chip
Xcode : 15.0
Xcode >  Product > Destination > Destination Architectures > Show Rosseta destination
